app/data_providers/curation_api_client.py

```diff
@@ -0,0 +1,46 @@
+import uuid
+from abc import ABC, abstractmethod
+from typing import List
+
+import requests
+import json
+from datetime import date
+
+from app.graphql.corpus_item import CorpusItem
+from app.models.corpus_item_model import CorpusItemModel
+from app.models.ranked_corpus_items_instance import RankedCorpusItemsInstance
+
+class CurationAPIFetchable(ABC):
+    @abstractmethod
+    async def get_ranked_corpus_slate(self, corpus_id: str, start_date: str, user_id) -> List[CorpusItem]:
+        return NotImplemented
+
+class CurationAPIClient(CurationAPIFetchable):
+    async def get_ranked_corpus_slate(self, corpus_id: str, start_date: str=None, user_id=None) -> List[CorpusItem]:
+        ranked_corpus_items_id = "NEW_TAB_EN_US"
+        if not start_date:
+            start_date = date.today().strftime("%Y-%m-%d")
+
+        request_headers = {
+            "apollographql-client-name": "recommendations-api",
+            "apollographql-client-version": "1"
+        }
+
+        query = f"""
+        query RecsApiItemRequest {{
+            scheduledSurface(id: "{ranked_corpus_items_id}") {{
+                id
+                items(date: "{start_date}") {{
+                  id
+                }}
+            }}        
+        }}
+        """
```

Comment:
Whenever data is inserted in code, it's good practice to escape that data, to prevent the interpreter from treating a malformed string as code. Injection vulnerability is a low risk here for several reasons, but it's still the right thing to do. I wouldn't rely on variables not being under user-control, validation happening elsewhere, or that there aren't any harmful GraphQL queries that can be run in the graph. All of those might change and are hard to verify.

1. The most reliable way to escape parameters is to let a trusted library do it for us. GraphQL support is pretty new in Python, so I'm assuming you use `requests` because there isn't a great client library available yet.
2. The next best thing is to call an escape function from a trusted library designed to escape in the given context: GraphQL strings. I couldn't immediately find one.
3. That leaves doing some custom validation/escaping ourselves. I'm reasonably confident that `json.dumps` would do the job. I'm also fine with regex validation or a Python date format function, along with a comment to the effect that this safeguards against GraphQL query injection.
```suggestion
            # Use json.dumps() to escape GraphQL strings, because we couldn't find one designed for GraphQL.
            # Better would be to find a trusted library that handles inserting parameters.
            scheduledSurface(id: {json.dumps(str(ranked_corpus_items_id))}) {{
                id
                items(date: {json.dumps(str(start_date))}) {{
                  id
                }}
            }}        
        }}
```
^ Untested code.

Reply:
Went ahead with your approach here! Extracted variables with names to make it clear why we're doing this.